backend/src/main/java/fk/prof/backend/model/request/CompositeByteBufInputStream.java

```diff
@@ -0,0 +1,110 @@
+package fk.prof.backend.model.request;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.Unpooled;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class CompositeByteBufInputStream extends InputStream {
+  private final CompositeByteBuf buffer;
+  private int readerIndexAtMark = 0;
+
+
+  public CompositeByteBufInputStream() {
+    buffer = Unpooled.compositeBuffer();
+  }
+
+  public CompositeByteBufInputStream(int maxNumComponents) {
+    buffer = Unpooled.compositeBuffer(maxNumComponents);
+  }
+
+  public void accept(ByteBuf newSource) {
+    buffer.addComponent(true, newSource);
+  }
+
+  @Override
+  public int read() throws IOException {
+    if (!buffer.isReadable()) {
+      return -1;
+    }
+    int readByte = buffer.readByte() & 0xff;
+    return readByte;
+  }
+
+  @Override
+  public int read(byte[] b, int off, int len) throws IOException {
+    int available = available();
+    if (available == 0) {
+      return -1;
+    }
+
+    len = Math.min(available, len);
+    buffer.readBytes(b, off, len);
+    return len;
+  }
+
+  @Override
+  public long skip(long n) throws IOException {
+    if (n > Integer.MAX_VALUE) {
+      return skipBytes(Integer.MAX_VALUE);
+    } else {
+      return skipBytes((int) n);
+    }
+  }
+
+  public int skipBytes(int n) throws IOException {
+    int nBytes = Math.min(available(), n);
+    buffer.skipBytes(nBytes);
+    return nBytes;
+  }
+
+  @Override
+  public int available() throws IOException {
+    return buffer.readableBytes();
+  }
+
+  @Override
+  public boolean markSupported() {
+    return true;
+  }
+
+  /**
+   * Besides marking the current position, this method discards all the bytes read up until this point.
+   * Calling reset does not restore the discarded bytes
+   * Call mark() when you are sure you have read all previous bytes correctly and can be freed up
+   *
+   * Readlimit parameter is accepted as arg to comply with Inputstream abstract class but is not used or supported by this impl
+   * @param readlimit
+   */
+  @Override
+  public void mark(int readlimit) {
```

Comment:
This to me seems too cryptic. I'd just create another method called discardReadBytes() and call it directly in relevant call-sites, and return false in markSupported.

This is not really mark-behavior anyone would anticipate. We don't have to build this over a bad interface like mark/reset.

Do we need marking for some other reason?

Usually interfaces are build to handle non-mark-supporting streams, because all socket-streams don't support it anyway (only file based streams do).

Let us discuss this f2f if necessary.